tokenserver/tests/test_service.py

```diff
@@ -806,8 +831,14 @@ def setUp(self):
         self.backend._safe_execute('delete from users')
         # Ensure the necessary service exists in the db.
         self.backend.add_service('sync-1.1', '{node}/1.1/{uid}')
+        self.backend.add_service('sync-1.5', '{node}/1.5/{uid}')
+        self.backend.add_service(
+            'spanner', 'https://spanner.example.com/1.5/{uid}')
```

Comment:
Hrm. I'm surprised to see this working when using "spanner" as the service name here. Since the spanner node is "just another sync storage node" I would expect these tests to all use `"sync-1.5"` as the service name, and for you to be adding the spanner node as `self.backend.add_node("sync-1.5", "https://spanner.example.com/1.5/{uid}")`.

Reply:
The "service name" parameter is really weird in this codebase, because it's a legacy hangover from a time when we thought we'd have multiple different services relying on Tokenserver for authentication. In practice we've only ever had one service, "sync-1.5", and we should expect to only ever have that one service in the future. If it wasn't such a wide-ranging change I'd suggest we go through and remove the `service` argument to all calls into the node-assignment backend, since it's totally useless to us in practice.

Reply:
Ah, to be honest, I wasn't quite sure what service actually equated to. I could see there being a new "spanner" service which doesn't have a node per se, since, in theory, spanner can handle an infinite amount of entries. I'm actually wondering how this might play out for ops. 

I could see a far future where we have a dedicated 'spanner' service or have clients `opt-in` themselves, but, I agree, this might be more effort than is wise.

Reply:
Ah, apparently, with the unified test suite we run, the MySQL side will fail if a "spanner" service is not defined, even though it may never call "spanner". Rather than add additional complexity to the test or functional code, I'll just add a comment to the `setUp()` function justifying why it's present.

Reply:
> the MySQL side will fail if a "spanner" service is not defined, even though it may never call "spanner".

I can't imagine what could be causing this in practice, but I can't argue with "makes the tests fail" :-)

Reply:
Moving the test to sql only seems to solve this, so happily dropping the nonce calls.